lib/pbench/test/functional/server/test_put.py

```diff
@@ -266,6 +261,33 @@ def test_index_all(self, server_client: PbenchServerClient, login_user):
         ), f"Didn't find all expected datasets, found {count} of {len(tarball_names)}"
 
 
+class TestElastic:
+    @pytest.mark.dependency(name="detail", depends=["index"], scope="session")
+    def test_details(self, server_client: PbenchServerClient, login_user):
+        """Check access to indexed Elasticsearch data
+
+        Perform a GET /datasets/details/{id} to be sure that basic run data
+        has been indexed into Elasticsearch and is available.
+        """
+        datasets = server_client.get_list(
+            metadata=["dataset.metalog.pbench,server.archiveonly"], owner="tester"
+        )
+        for d in datasets:
+            print(f"\t... checking run index for {d.name}")
+            response = server_client.get(
+                API.DATASETS_DETAIL, {"dataset": d.resource_id}, json={}
+            )
+            detail = response.json()
+            indexed = not d.metadata["server.archiveonly"]
+            if indexed:
+                assert (
+                    d.metadata["dataset.metalog.pbench"]["script"]
+                    == detail["runMetadata"]["script"]
+                )
+            else:
+                print(f"\t\t... {d.name} was not indexed")
```

Comment:
I don't think it's ambiguous at all. We can't check the indexed data because there's no index. With that said, I suppose I could go for the more direct alternative.